File: src/ManagementPlatform.Domain/Tenant.cs
namespace ManagementPlatform.Domain;

public sealed class Tenant
{
    public long Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public DateTimeOffset CreatedAt { get; set; }

    public List<Order> Orders { get; set; } = [];
}

File: src/main/java/com/managementplatform/domain/model/Tenant.java
package com.managementplatform.domain.model;

import java.time.Instant;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public final class Tenant {
    private final long id;
    private final String name;
    private final String email;
    private final Instant createdAt;
    private final List<Order> orders = new ArrayList<>();

    public Tenant(long id, String name, String email, Instant createdAt) {
        this.id = id;
        this.name = name;
        this.email = email;
        this.createdAt = createdAt;
    }

    public long id() {
        return id;
    }

    public String name() {
        return name;
    }

    public String email() {
        return email;
    }

    public Instant createdAt() {
        return createdAt;
    }

    public List<Order> orders() {
        return Collections.unmodifiableList(orders);
    }

    public void addOrder(Order order) {
        orders.add(order);
    }
}
